internal/checkconfig/checkconfig.go

```diff
@@ -170,13 +170,24 @@ func LoadConfig(configPath string) (*Config, error) {
 				if !ok {
 					log.Println("Didn't find 'kubeconfig' in 'data-gatherers.k8s/pods' configuration. Assuming it runs in-cluster.")
 				}
+				excludedNamespaces := []string{}
+				excludedNamespacesFromFile, ok := dataGathererConfigMap["exclude-namespaces"].([]interface{})
```

Comment:
🤔 do we not have any typing on this config file we are decoding into...? It may not be specific to this PR, but it seems totally wrong that we're casting to a slice of interfaces here - is it not possible for us to decode into the structure we actually earlier on than this, and actually make use of the type system?

Reply:
So this is just the config loading for `check` which is being deprecated. It's not great but we decided not to work on it as it's going away. The `agent` part which replaces it has much better config loading.